### PR TITLE
OKTA-645236 : G3 : Adding disableAutocomplete compatibility in password enroll/reset flow

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## Description:

The purpose of this PR is to add compatibility for the newly added `features.disableAutocomplete` config option for Password Enroll/Reset flows. 

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-645236](https://oktainc.atlassian.net/browse/OKTA-645236)

### CODE REVIEW

File: src/v3/src/transformer/profile/transformEnrollProfile.ts [hunk 1]
```diff
@@ -67,7 +72,7 @@ export const transformEnrollProfile: IdxStepTransformer = ({ transaction, formBa
       ...passwordElement.options,
       attributes: {
         ...passwordElement.options?.attributes,
-        autocomplete: 'new-password',
+        autocomplete: disableAutocomplete ? 'off' : 'new-password',
```
Comment (shawyu-okta) [inline src/v3/src/transformer/profile/transformEnrollProfile.ts:75]: Should we consider a separate transformer that overwrites this for us based on `disableAutocomplete`? Thinking of something like the LTR transformer @leemchale-okta just added

Comment (glenfannin-okta) [reply src/v3/src/transformer/profile/transformEnrollProfile.ts:75]: Oh yes this is a good idea 🙏

File: src/v3/src/components/PhoneAuthenticator/PhoneAuthenticator.tsx [hunk 2]
```diff
@@ -147,7 +148,7 @@ const PhoneAuthenticator: UISchemaElementComponent<UISchemaElementComponentWithV
           }}
           inputProps={{
             'data-se': 'extension',
-            autocomplete: 'tel-extension',
+            autocomplete: disableAutocomplete ? 'off' : 'tel-extension',
```
Comment (shawyu-okta) [inline src/v3/src/components/PhoneAuthenticator/PhoneAuthenticator.tsx:151]: Ah too bad this falls outside our pattern

**Review by shawyu-okta — APPROVED** [review]
nice work!

**Pull request merged**